File: packages/expo-kline-chart/android/src/main/java/expo/modules/klinechart/KLineModels.kt
package expo.modules.klinechart

import android.graphics.Color

// MARK: - Data Models

data class KLineDataItem(
    val timestamp: Double,
    val open: Double,
    val high: Double,
    val low: Double,
    val close: Double,
    val volume: Double
)

data class ChartConfig(
    // Binance color scheme
    var upColor: Int = Color.parseColor("#0ECB81"),
    var downColor: Int = Color.parseColor("#F6465D"),
    var gridColor: Int = Color.parseColor("#3A4250"),
    var textColor: Int = Color.parseColor("#848E9C"),
    var crosshairColor: Int = Color.parseColor("#5E6673"),
    var backgroundColor: Int = Color.parseColor("#0B0E11"),
    var maColors: List<Int> = listOf(
        Color.parseColor("#F0B90B"),
        Color.parseColor("#6149CD"),
        Color.parseColor("#2196F3")
    ),
    var isDarkMode: Boolean = true,
    var candleSpacing: Float = 1.5f,
    var minCandleWidth: Float = 2f,
    var maxCandleWidth: Float = 30f,
    var initialCandleWidth: Float = 8f
) {
    // Theme-aware overlay colors (matching iOS KLineModels.swift)
    val priceLineColor: Int get() = if (isDarkMode) Color.WHITE else Color.parseColor("#1E2329")
    val priceLabelBorderColor: Int get() = Color.parseColor("#B7BDC6")
    val priceLabelChevronColor: Int get() = if (isDarkMode) Color.parseColor("#EAECEF") else Color.parseColor("#474D57")
    val priceLabelTextColor: Int get() = if (isDarkMode) Color.parseColor("#F0F1F2") else Color.parseColor("#1E2329")
    val priceLabelBgColor: Int get() = if (isDarkMode) setAlpha(Color.parseColor("#0B0E11"), 140) else setAlpha(Color.parseColor("#FFFFFF"), 191)

    val crosshairLabelBgColor: Int get() = if (isDarkMode) Color.parseColor("#E6E8EA") else Color.parseColor("#1E2329")
    val crosshairLabelTextColor: Int get() = if (isDarkMode) Color.parseColor("#1E2329") else Color.parseColor("#FFFFFF")

    val tooltipBgColor: Int get() = if (isDarkMode) setAlpha(Color.parseColor("#0B0E11"), 166) else setAlpha(Color.parseColor("#FFFFFF"), 217)
    val tooltipBorderColor: Int get() = if (isDarkMode) Color.parseColor("#2B3139") else Color.parseColor("#EAECEF")
    val tooltipLabelColor: Int get() = Color.parseColor("#848E9C")
    val tooltipValueColor: Int get() = if (isDarkMode) Color.parseColor("#EAECEF") else Color.parseColor("#1E2329")

    val highLowMarkerColor: Int get() = if (isDarkMode) Color.WHITE else Color.parseColor("#1E2329")

    private fun setAlpha(color: Int, alpha: Int): Int {
        return Color.argb(alpha, Color.red(color), Color.green(color), Color.blue(color))
    }
}

data class YScaleInfo(
    val min: Double,
    val max: Double,
    val range: Double,
    val pixelsPerUnit: Float
)

File: packages/expo-kline-chart/ios/KLineModels.swift
import UIKit

// MARK: - Data Models

struct KLineDataItem {
    let timestamp: Double
    let open: Double
    let high: Double
    let low: Double
    let close: Double
    let volume: Double
}

struct ChartConfig {
    // Binance color scheme
    var upColor: UIColor = UIColor(hex: "#0ECB81")       // Binance green
    var downColor: UIColor = UIColor(hex: "#F6465D")      // Binance red
    var gridColor: UIColor = UIColor(hex: "#3A4250")      // Grid lines
    var textColor: UIColor = UIColor(hex: "#848E9C")      // Gray text
    var crosshairColor: UIColor = UIColor(hex: "#5E6673")
    var backgroundColor: UIColor = UIColor(hex: "#0B0E11") // Binance dark bg
    var maColors: [UIColor] = [
        UIColor(hex: "#F0B90B"),  // Binance yellow (MA5)
        UIColor(hex: "#6149CD"),  // Purple (MA10)
        UIColor(hex: "#2196F3"),  // Blue (MA20)
    ]
    var isDarkMode: Bool = true
    var candleSpacing: CGFloat = 1.5
    var minCandleWidth: CGFloat = 2
    var maxCandleWidth: CGFloat = 30
    var initialCandleWidth: CGFloat = 8

    // MARK: - Theme-aware overlay colors

    var priceLineColor: UIColor { isDarkMode ? .white : UIColor(hex: "#1E2329") }
    var priceLabelBorderColor: UIColor { UIColor(hex: "#B7BDC6") }
    var priceLabelChevronColor: UIColor { isDarkMode ? UIColor(hex: "#EAECEF") : UIColor(hex: "#474D57") }
    var priceLabelTextColor: UIColor { isDarkMode ? UIColor(hex: "#F0F1F2") : UIColor(hex: "#1E2329") }
    var priceLabelBgColor: UIColor { isDarkMode ? UIColor(hex: "#0B0E11").withAlphaComponent(0.55) : UIColor(hex: "#FFFFFF").withAlphaComponent(0.75) }

    var crosshairLabelBgColor: UIColor { isDarkMode ? UIColor(hex: "#E6E8EA") : UIColor(hex: "#1E2329") }
    var crosshairLabelTextColor: UIColor { isDarkMode ? UIColor(hex: "#1E2329") : UIColor(hex: "#FFFFFF") }

    var tooltipBgColor: UIColor { isDarkMode ? UIColor(hex: "#0B0E11").withAlphaComponent(0.65) : UIColor(hex: "#FFFFFF").withAlphaComponent(0.85) }
    var tooltipBorderColor: UIColor { isDarkMode ? UIColor(hex: "#2B3139") : UIColor(hex: "#EAECEF") }
    var tooltipLabelColor: UIColor { UIColor(hex: "#848E9C") }
    var tooltipValueColor: UIColor { isDarkMode ? UIColor(hex: "#EAECEF") : UIColor(hex: "#1E2329") }

    var highLowMarkerColor: UIColor { isDarkMode ? .white : UIColor(hex: "#1E2329") }
}

struct YScaleInfo {
    let min: Double
    let max: Double
    let range: Double
    let pixelsPerUnit: CGFloat
}
